pages/docs/tutorials/native/dynamic-libraries.md

```diff
@@ -0,0 +1,339 @@
+Swift, and integrates with their reference counters. You may want to find more details
+in the [documentation](https://github.com/JetBrains/kotlin-native/blob/master/OBJC_INTEROP.md)
+or to check the related [TUTORIAL_KOTLIN_macOS_FRAMEWORK_LINK].
+
+### Our Library Functions
+
+Let's take a look on the nests structure with path `kotlin.root.demo`. The path 
+mimics the package structure of our Kotlin code with `kotlin.root.` prefix.
+
+There is `kotlin.root.demo.DemoClazz` field, that 
+represents the `DemoClazz` from Kotlin. The `DemoClazz#foo` is
+represented with the `foo` field. The only difference is that 
+the `foo` accepts `this` reference as the first parameter. 
+C language does not support objects and that is the reason to pass
+`this` pointer explicitly.
+
+There is constructor the `DemoClazz` field (aka `kotlin.root.demo.DemoClazz.DemoClazz`),
+which is the constructor function to create an instance of the `DemoClazz`.
+
+Properties are translated into functions too. The `get_` and `set_` prefix
+is used to name the getter and the setter functions respectively. For example, 
+the readonly property `globalString` from Kotlin is turned 
+into the `get_globalString` function in C. 
+
+Global functions `ints`, `floats` or `strings` are turned into functions pointers in
+the `kotlin.root.demo` anonymous struct.
+ 
+### The Entry Point
+
+We see how API is created. To start with, we need to initialize the 
+`demo_ExportedSymbols` structure. Let's take a look at the latest part 
+of the `demo_api.h` for that:
+
+```c
+extern demo_ExportedSymbols* demo_symbols(void);
+```
+
+The function `demo_symbols` allows us to open the door from a native 
+code to Kotlin/Native library. That is the entry point we use. The 
+library name is used as the prefix for the function name. 
+
+
+Kotlin/Native object references does not support multi-threaded access. It
+might be necessary to host the returned `demo_ExportedSymbols*` pointer
+per thread.
+
+## Using Generated Headers from C
+
+The usage from C not complicated and strait forward. We create a `main.c` file with the following 
+code: 
+```c
+#include "demo_api.h"
+#include "stdio.h"
+
+int main(int argc, char** argv) {
+  //initialize Kotlin/Native library
+  demo_ExportedSymbols* lib = demo_symbols();
+
+  //call functions
+  lib->kotlin.root.demo.ints(1, 2, 3, 4);
+  lib->kotlin.root.demo.floats(1.0f, 2.0);
+
+  //use C and Kotlin/Native strings
+  const char* str = "Hello from Native!";
+  const char* response = lib->kotlin.root.demo.strings(str);
+  printf("in: %s\nout:%s\n", str, response);
+  lib->DisposeString(response);
+
+  //create Kotlin object instance
+  demo_kref_demo_DemoClazz newInstance 
+          = lib->kotlin.root.demo.DemoClazz.DemoClazz();
+  long result = lib->kotlin.root.demo.DemoClazz.foo(newInstance, 42);
+  printf("DemoClazz returned %ld\n", result);
+  lib->DisposeStablePointer(newInstance.pinned);
+
+  return 0;
```

Comment:
Following things need to be mentioned:
 - Only operations explicitly mentioned in the source code are exposed to the C world
 - If you're using multiple Kotlin/Native libraries in the same C program object handles of those are generally incompatible, and shall not be passed each to another.

Reply:
Could you please elaborate on the first. What else, beyond `DisposeStablePointer`, `DisposeString` and `IsInstance` it may generate?